lib/plugins/bodyParser.js

```diff
@@ -192,6 +187,14 @@ function bodyParser(options) {
                 break;
         }
 
+        // if still no match, do more expensive regex matches. map any +json to
+        // application/json.
+        if (!parser) {
```

Comment:
Nit/preference: the above seemed more obvious to me. This nested `if` required me to think a lot more about what was happening. I think it is handling both the explicit match and the regex as separate cases. Would it make sense to remove the `application/json` case above and assume that if `jsonPatternMatcher` matches here that `parser` should always be set to `parseJson`? This is similar to what the previous code did by coercing type into `application/json` if the regex matched.

Reply:
Good question! The primary motivator for the PR was to avoid running/creating the regex when it wasn't necessary (since we were lifting the regex too to also avoid creating a new one for every request). The string comparisons are cheaper and likely to cover the 90% use case, so I wanted to reserve running the regex matching only when those cases failed. I expanded the comment with this context.